mantle/cmd/plume/release.go

```diff
@@ -186,9 +186,11 @@ func makeReleaseAMIsPublic(rel release.Release) bool {
 		}
 	}
 
-	// if none passed, then it's likely a more fundamental issue like wrong
-	// permissions or API usage, etc... let's just hard fail in that case
-	if !at_least_one_passed {
+	// If none passed, then it's likely a more fundamental issue like wrong
+	// permissions or API usage, etc... let's just hard fail in that case. But
+	// if also not even one failed, it must mean that aren't any AMIs in the
+	// release metadata. In that case, we no-op.
+	if !at_least_one_passed && !at_least_one_failed {
```

Comment:
I hit this in a testing a dev pipeline where I had purposely disabled almost everything.

> If we are checking for "must mean that aren't any AMIs in the release metadata" then maybe let's just check that explicitly earlier, which I think would make this code easier to follow.

I think if it were a simple `len` check, it'd be nice. But it requires nested loops and although performance doesn't matter here, it feels really awkward to write the same boilerplate-y nested loops twice. Maybe we can rename it to something else. Any ideas?

Reply:
good point. This isn't as easy because you have to iterate through the arches to find if there are any AWS AMIs.

My only suggestion really would be to have a `at_least_one_tried` var that gets set to true after we get past the `awsmedia == nil` check. then we could have a separate check that more clearly illustrates the "no-op" case.

Reply:
Nice update with `at_least_one_tried`. It takes me much less time to comprehend things now.